metadata-ingestion/src/datahub/ingestion/source/sql_queries.py

```diff
@@ -52,6 +58,33 @@
 logger = logging.getLogger(__name__)
 
 
+class TrackingSchemaResolver:
+    """Wrapper around SchemaResolver that tracks cache hits and misses for performance monitoring."""
+
+    def __init__(
+        self, schema_resolver: SchemaResolver, report: "SqlQueriesSourceReport"
+    ):
+        self._schema_resolver = schema_resolver
+        self._report = report
+
+    def __getattr__(self, name):
+        """Delegate all other attributes to the wrapped schema resolver."""
+        return getattr(self._schema_resolver, name)
+
+    def resolve_table(self, table):
+        """Resolve table with cache hit/miss tracking."""
+        result = self._schema_resolver.resolve_table(table)
+
+        # Check if this was a cache hit or miss by looking at the cache
+        urn = self._schema_resolver.get_urn_for_table(table)
+        if urn in self._schema_resolver._schema_cache:
+            self._report.num_schema_cache_hits += 1
+        else:
+            self._report.num_schema_cache_misses += 1
+
+        return result
```

Comment:
This functionality could be moved to the `SchemaResolver` class itself, as well as a `SchemaResolverReport` class may be defined to hold these two counters.